### PR TITLE
Fix Unit Tests for VaccineStatusService AB#11648

### PR DESCRIPTION
**Pull request by pagustinqt20**
# Fix Unit Tests for VaccineStatusService [AB#11648](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/11648)

-   [x] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

Fix Unit Test for Vaccine Status Service based on the new implemented way to get the PDF for both public and authenticated.

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [x] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

NO

### Browsers Tested

-   [ ] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

-   Fulfills Work Item Requirements

    -   The changes meet/implement story's acceptance criteria. Fixes identified problems if bug.
    -   Changes not directly related to the task requirements need to be documented on the PR.

-   Compilation / Tests

    -   The changes do not break compilation and/or unit or functional tests; unless other item addresses them specifically.

-   Logic Problems / Functional Behaviour

    -   The changes work as intended and do not introduce problems.

-   Performance

    -   The changes do not introduce obvious performance issues.

-   Documented Standards

    -   The changes adhere to the team's documented [Coding Standards](https://github.com/bcgov/healthgateway/wiki/standards).

-   Readability / Maintainability
    -   The changes can be easily understood/read and allow for future enhancements without major refactoring. Readability is preferred over "clever code".
    -   Reasoning for changes needs to be clearly articulated. Disagreements should be arbitrated by a third developer.

### CODE REVIEW

**Review by BrianMaki — APPROVED** [review]

File: Apps/Immunization/test/unit/Services.Test/VaccineStatusServiceTests.cs [hunk 1]
```diff
@@ -40,21 +45,26 @@ namespace HealthGateway.Immunization.Test.Services
     [System.Diagnostics.CodeAnalysis.SuppressMessage("Major Code Smell", "S125:Sections of code should not be commented out", Justification = "Ignore broken tests")]
```
Comment (MikeLyttle) [inline Apps/Immunization/test/unit/Services.Test/VaccineStatusServiceTests.cs:45]: This line should be removed now that we are not disabling any tests by commenting them out.

**Review by sslaws — APPROVED** [review]

**Review by MikeLyttle — CHANGES_REQUESTED** [review]
Review-in-progress.

File: Apps/Immunization/test/unit/Services.Test/VaccineStatusServiceTests.cs [hunk 2]
```diff
@@ -99,15 +110,295 @@ public void ShouldGetVaccineStatus()
...
+                mockDelegate.Object,
+                GetMemoryCache(),
+                this.GetHttpContextAccessor().Object);
+
+            var actualResult = Task.Run(async () => await service.GetAuthenticatedVaccineStatus(this.hdid).ConfigureAwait(true)).Result;
             Assert.True(expectedResult.IsDeepEqual(actualResult));
         }
 
+        /// <summary>
+        /// GetVaccineStatus - when the status when there is data mismatch.
```
Comment (MikeLyttle) [inline Apps/Immunization/test/unit/Services.Test/VaccineStatusServiceTests.cs:187]: The comment should reference `GetPublicVaccineStatus` and the description should be updated to be clearer.

File: Apps/Immunization/test/unit/Services.Test/VaccineStatusServiceTests.cs [hunk 3]
```diff
@@ -99,15 +110,295 @@ public void ShouldGetVaccineStatus()
...
+        /// </summary>
+        [Fact]
+        public void ShouldGetErrorDismatchVaccineStatus()
```
Comment (MikeLyttle) [inline Apps/Immunization/test/unit/Services.Test/VaccineStatusServiceTests.cs:190]: `Dismatch` should be `DataMismatch`.

File: Apps/Immunization/test/unit/Services.Test/VaccineStatusServiceTests.cs [hunk 4]
```diff
@@ -99,15 +110,295 @@ public void ShouldGetVaccineStatus()
...
+                mockAuthDelegate.Object,
+                mockDelegate.Object,
+                GetMemoryCache(),
+                this.GetHttpContextAccessor().Object);
+
+            string dobString = this.dob.ToString("yyyy-MM-dd", CultureInfo.CurrentCulture);
+            string dovString = this.dov.ToString("yyyy-MM-dd", CultureInfo.CurrentCulture);
+            var actualResult = Task.Run(async () => await service.GetPublicVaccineStatus(this.phn, dobString, dovString).ConfigureAwait(true)).Result;
+            Assert.Equal(Common.Constants.ResultType.ActionRequired, actualResult.ResultStatus);
+            Assert.Equal(expectedResult.ResultError.ActionCode, actual: actualResult.ResultError?.ActionCode);
```
Comment (MikeLyttle) [inline Apps/Immunization/test/unit/Services.Test/VaccineStatusServiceTests.cs:249]: I don't think using a named parameter (`actual:`) is required here.  There are 3 places in the file where `actual:` can be removed.

File: Apps/Immunization/test/unit/Services.Test/VaccineStatusServiceTests.cs [hunk 5]
```diff
@@ -99,15 +110,295 @@ public void ShouldGetVaccineStatus()
...
+        }
+
+        /// <summary>
+        /// GetPublicVaccineProof - Happy path.
+        /// </summary>
+        [Fact]
+        public void ShouldPublicGetVaccineProof()
```
Comment (MikeLyttle) [inline Apps/Immunization/test/unit/Services.Test/VaccineStatusServiceTests.cs:256]: Method name should be `ShouldGetPublicVaccineProof`

**Review by MikeLyttle — CHANGES_REQUESTED** [review]
We may also want to add unit tests to handle the "RefreshInProgress" scenario and the "NotFound" result, but that can be done in a separate task.

**Comment by pagustinqt20** [comment]
@MikeLyttle  Will add unit test for "RefreshInProgress" scenario and the "NotFound" I can include it in this change. It is a good idea to add those

**Comment by pagustinqt20** [comment]
@MikeLyttle  Added new tests for NotFound and RefreshInprogress. I added parameter for isPublicEndpoint.

**Review by MikeLyttle — APPROVED** [review]

**Review by BrianMaki — APPROVED** [review]

**Pull request merged**